workflow-Estimate/src/main/java/ca/on/oicr/pde/workflows/EstimateWorkflow.java

```diff
@@ -0,0 +1,247 @@
+        
+        Job preProcess = postProcessRSEM(this.inputRSEMFiles, this.inputSTARFiles, postProcessedRSEM);
+        parentJob = preProcess;
+        
+        Job runEstimate = launchEstimate(postProcessedRSEM);
+        runEstimate.addParent(parentJob);
+        parentJob = runEstimate;
+        
+        // Provision out estimae and ssgsea outputs
+        SqwFile estimateOutput = createOutputFile(estimateGCT, TXT_METATYPE, this.manualOutput);
+        estimateOutput.getAnnotations().put("estimate_output", "estimate");
+        parentJob.addFile(estimateOutput);
+        
+        SqwFile ssGSEAOutput = createOutputFile(ssGSEA, TXT_METATYPE, this.manualOutput);
+        ssGSEAOutput.getAnnotations().put("ssgsea_output", "ssgsea");
+        parentJob.addFile(ssGSEAOutput);
+//        
+    }
+    
+    private Job launchEstimate(String inRSEM) {
+        Job runEst = getWorkflow().createBashJob("run_estimate_ssgsea");
+        Command cmd = runEst.getCommand();
+        cmd.addArgument("export R_LIBS=" + rLib + ";");
+        cmd.addArgument(this.rScript);
+        cmd.addArgument(this.estimateScript);
+        cmd.addArgument(inRSEM);
+        cmd.addArgument(this.dataDir);
+        cmd.addArgument(this.gmtFile);
+        cmd.addArgument(this.ensFile);
+        cmd.addArgument(getWorkflowBaseDir() + "/dependencies/convert_rsem_results_zscore.r");
+        runEst.setMaxMemory(Integer.toString(this.estimateMem * 1024));
+        runEst.setQueue(getOptionalProperty("queue", ""));
+        return runEst;
+    }   
+    
+    private Job postProcessRSEM(String inRSEMs, String inSTARs, String postProcessedRSEM) {
+        Job postProcessRSEMGeneCounts = getWorkflow().createBashJob("post_process_RSEM");
+        Command cmd = postProcessRSEMGeneCounts.getCommand();
+        Map<String, List<String>> map = this.getRsemStarMap(inRSEMs, inSTARs);
+        for (String key: map.keySet()){
+//            Log.debug(key + " ... " + map.get(key));
+            String geneCount = this.tmpDir + key + ".count";
+            String geneRcount = this.tmpDir + key + ".rcount";
+            String gene = getFiles().get("RSEM_"+key).getProvisionedPath();
+            String rtab = getFiles().get("STAR_"+key).getProvisionedPath();
+            cmd.addArgument("echo \"" + key + "\" > " + geneCount + ";");
+            cmd.addArgument("cut -f5 " + gene + " | awk 'NR>1' >> " + geneCount + ";");
+            cmd.addArgument("echo \"" + key + "\" > " 
+                    + geneRcount 
+                    + ";");
+            cmd.addArgument("awk 'NR>4 {if ($4 >= $3) print $4; else print $3}' " 
+                    + rtab + " >> " + geneRcount + ";");
+            cmd.addArgument("cp " + rtab + " " + this.tmpDir + ";");
+        }
+        cmd.addArgument("STARG=`ls " + this.tmpDir + "*.tab | head -1`;");
+        cmd.addArgument("if [ ! -z $STARG ]; then awk 'NR>3 {print $1}' $STARG | sed \"s/N\\_ambiguous/gene\\_id/\" > " + this.tmpDir + "sgene; fi;");
```

Comment:
Hopefully this works. We occasionally have escaping problems when we pass strings in like this.